src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs

```diff
@@ -26,12 +26,17 @@ private AddPropertyTransform<CommonCompilerOptions>[] EmitEntryPointTransforms
         private AddPropertyTransform<CommonCompilerOptions>[] KeyFileTransforms
             => new []
             {
-                new AddPropertyTransform<CommonCompilerOptions>("AssemblyOriginatorKeyFile",
+                new AddPropertyTransform<CommonCompilerOptions>("KeyOriginatorFile",
                     compilerOptions => compilerOptions.KeyFile,
                     compilerOptions => !string.IsNullOrEmpty(compilerOptions.KeyFile)),
                 new AddPropertyTransform<CommonCompilerOptions>("SignAssembly",
                     "true",
                     compilerOptions => !string.IsNullOrEmpty(compilerOptions.KeyFile))
+                    .WithMSBuildCondition(" '$(OS)' == 'Windows_NT' "),
+                new AddPropertyTransform<CommonCompilerOptions>("PublicSign", 
+                    "true",
+                    compilerOptions => !string.IsNullOrEmpty(compilerOptions.KeyFile))
```

Comment:
Should also include a condition if publicSign isn't explicitly defined in the project.json.